pipeline/artifactory/artifactory.go

```diff
@@ -2,24 +2,13 @@
 package artifactory
 
 import (
-	"bytes"
 	"encoding/json"
 	"fmt"
-	"html/template"
-	"io"
 	"io/ioutil"
 	"net/http"
-	"net/url"
-	"os"
-	"strings"
 
-	"github.com/apex/log"
-	"github.com/pkg/errors"
-	"golang.org/x/sync/errgroup"
-
-	"github.com/goreleaser/goreleaser/config"
 	"github.com/goreleaser/goreleaser/context"
-	"github.com/goreleaser/goreleaser/internal/artifact"
+	internal "github.com/goreleaser/goreleaser/internal/httpupload"
```

Comment:
why not call it `httpupload`? `internal` doesn't seem to be a descriptive name..

Reply:
Yep...
It all started when needed to import internal/httpupload from pipeline/httpupload, there the alias was required, calling it internal seemed like a good choice... from there it just spilled over pipeline/artifactory because it seemed more uniform/readable to call it the same from the two client packages.
I must say I would really like to toss the "httpupload" name completely but couldn't come up with something better.
I also believe it would be best to call the internal package something different to the pipeline packages to indicate the difference in generality, but not just "http" because collides with "net/http"...

Maybe something like this:

![](http://www.plantuml.com/plantuml/svg/SoWkIImgAStDuOhEp2j9BSfBpD7FA2afA2s8LR3HrLM8BiWiICt9p4lL1ohBJ4jCBialgembgBYWj2IMoo4rBmLeCW00)

Although I didn't put too much thought on it, I never found good names for the internal and the  pipeline http packages.

Ideas?

Reply:
not really :/